### ISSUE TITLE
why Modsecurity3.0 use more than 10ms when processing one httprequest with cookie header

### ISSUE THREAD

**Issue by hbt19860104** [SpiderLabs/ModSecurity#1899]
Hi,

    I have a performance issue in Modsecurity3.0。 When i issue a http request with fiddler like below:

User-Agent: Fiddler
Host: hb.crm2.qq.com
Connection: keep-alive
Pragma: no-cache
Cache-Control: no-cache
Upgrade-Insecure-Requests: 1
DNT: 1
Accept: */*
Accept-Encoding: gzip, deflate
Accept-Language: zh-CN,zh;q=0.9
Cookie: pgv_pvid=5233419322; pac_uid=0_5b337de3e0fb3; pgv_pvi=4538145792; tvfe_boss_uuid=9a1f1024cd459f48; RK=TXTR4wME6K; ptcz=df196009c81904b7786af85347305f022fb47abe692ba27fbabbea6b2a723c78; pt2gguin=o0363214664; uin=o0363214664; skey=@biHRtLhaz; ptisp=ctc

modsecurity3.0 use about 13ms to deal with the request. I only use the following function:

msc_process_connection, msc_process_uri, msc_add_request_header, msc_process_request_headers,

msc_append_request_body, msc_process_request_body. And i use the stand OWASP ModSecurity Core Rule Set rules.

As you see, I only process the request data, and doesn't process the response data.

Is this situation normal, or there's some problem i using the modsecurity in my code?

Thanks you very much!

**Comment by zimmerle** [comment]
Hi @hbt19860104,

Regardless of using or not the request body we recommend you to call the libraries methods in order to maintain the full compatibility. 

How many **ms** it takes without the OWASP CRS?

I am assuming that you have wrote our own connector, without modsec, how many seconds the response is delivered?

**Comment by hbt19860104** [comment]
I did call the libraries methods of modsecurity. The 10ms or 13 ms i methioned above is the cost i call msc_process_request_body. I did call gettimeofday before and after msc_process_request_body, and calculate the duration. And yes, i wrote my own connector, and use libmodsecurity with OWASP CRS.
It cost about 140 microseconds of msc_process_request_body function without OWASP CRS.
I think it doesn't any matter with the response data, because i just simply delieved forward the packets without modsecurity  for response packets.
